ts/input/tex/TexParser.ts

```diff
@@ -66,6 +66,11 @@ export default class TexParser {
    */
   public currentCS: string = '';
 
+  /**
+   * A stack to save the string positions when we restart the parser.
+   */
+  private saveI: number[] = [0];
```

Comment:
If you make the changes I suggest to `saveI`, it should just be a number, here, not an array.

Reply:
Simplified `saveI`.